src/styles/main.scss

```diff
@@ -0,0 +1,77 @@
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+$calendar-day-width: 100px;
+
+.calendar {
+  width: $calendar-day-width * 7 + 6px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  $days: (
+    mon: 0,
+    tue: 1,
+    wed: 2,
+    thu: 3,
+    fri: 4,
+    sat: 5,
+    sun: 6,
+  );
+
+  @each $day, $i in $days {
+    &--start-day-#{$day} {
+      .calendar__day:nth-child(1) {
+        margin-left: (
+          $calendar-day-width * ($i) + ($calendar-day-width * ($i) / 100)
```

Comment:
The `margin-left` calculation for the first day of the week seems to be adding an extra percentage-based margin. Verify if this is intentional and aligns with your design intentions. If not, consider adjusting the calculation.